source/ada/lsp-ada_handlers.adb

```diff
@@ -1438,16 +1452,259 @@ package body LSP.Ada_Handlers is
+
+      procedure Parse (Node : Libadalang.Analysis.Ada_Node) is
+         use Libadalang.Analysis;
+      begin
+         if Node = No_Ada_Node then
+            return;
+         end if;
+--        Cat_Namespace,
+--        Cat_Constructor,
+--        Cat_Destructor,
+--        Cat_Structure,
+--        Cat_Case_Inside_Record,
+--        Cat_Union,
+--        Cat_Custom);            --  Custom construct
+
+         case Node.Kind is
+            when Ada_Package_Decl |
+                 Ada_Generic_Formal_Package |
+                 Ada_Package_Body |
+                 --  Cat_Package
+
+                 Ada_Type_Decl |
+
+                 Ada_Classwide_Type_Decl |
+--        Cat_Class
+
+                 Ada_Protected_Type_Decl |
+--        Cat_Protected
+
+                 Ada_Task_Type_Decl |
+                 Ada_Single_Task_Type_Decl |
+--        Cat_Task
+
+                 Ada_Subp_Decl |
+                 Ada_Subp_Body |
+                 Ada_Subp_Spec |
+                 Ada_Generic_Formal_Subp_Decl |
+                 Ada_Abstract_Subp_Decl |
+                 Ada_Abstract_Formal_Subp_Decl |
+                 Ada_Concrete_Formal_Subp_Decl |
+                 Ada_Generic_Subp_Internal |
+                 Ada_Null_Subp_Decl |
+                 Ada_Subp_Renaming_Decl |
+                 Ada_Subp_Body_Stub |
+                 Ada_Generic_Subp_Decl |
+                 Ada_Generic_Subp_Instantiation |
+                 Ada_Generic_Subp_Renaming_Decl |
+                 Ada_Subp_Kind_Function |
+                 Ada_Subp_Kind_Procedure |
+                 Ada_Access_To_Subp_Def |
+--        Cat_Procedure
+--        Cat_Function
+--        Cat_Method
+
+                 Ada_Case_Stmt |
+--        Cat_Case_Statement
+
+                 Ada_If_Stmt |
+--        Cat_If_Statement
+
+                 Ada_For_Loop_Stmt |
+                 Ada_While_Loop_Stmt |
+--        Cat_Loop_Statement
+
+                 Ada_Begin_Block |
+                 Ada_Decl_Block |
+--        Cat_Declare_Block
+--        Cat_Simple_Block
+
+--                 Ada_Return_Stmt |
+--                 Ada_Extended_Return_Stmt |
+                 Ada_Extended_Return_Stmt_Object_Decl |
+--        Cat_Return_Block
+
+                 Ada_Select_Stmt |
+--        Cat_Select_Statement
+
+                 Ada_Entry_Body |
+--        Cat_Entry
+
+                 Ada_Exception_Handler |
+--        Cat_Exception_Handler
+
+                 Ada_Pragma_Node_List |
+                 Ada_Pragma_Argument_Assoc |
+                 Ada_Pragma_Node |
+--        Cat_Pragma
+
+                 Ada_Aspect_Assoc_List |
+                 Ada_Aspect_Assoc |
+                 Ada_Aspect_Spec =>
+--        Cat_Aspect
+
+               Store_With;
+               foldingRange.kind :=
+                 (Is_Set => True, Value => LSP.Messages.Region);
+
+               Sloc := Sloc_Range (Data (Node.Token_Start));
+               foldingRange.startLine := LSP_Number (Sloc.Start_Line) - 1;
```

Comment:
Could you reuse `LSP.Lal_Utils.Get_Node_Location` function? This will simplifies job when we will fix Position.column calculation at some point. The offsets are based on a UTF-16 string representation, so counting it as character index is incorrect.